src/seq_analysis/orf.rs

```diff
@@ -137,28 +143,34 @@ where
             self.state.codon.push_back(*nuc.borrow());
             offset = (index + 1) % 3;
 
+            // check if entering orf
+            if self.finder.start_codons.contains(&self.state.codon) {
+                self.state.start_pos[offset].push(index);
+            }
             // inside orf
-            if self.state.start_pos[offset].is_some() {
+            if !self.state.start_pos[offset].is_empty() {
                 // check if leaving orf
                 if self.finder.stop_codons.contains(&self.state.codon) {
-                    // check if length is sufficient
-                    if index + 1 - self.state.start_pos[offset].unwrap() > self.finder.min_len {
-                        // build results
-                        result = Some(Orf {
-                            start: self.state.start_pos[offset].unwrap() - 2,
-                            end: index + 1,
-                            offset: offset as i8,
-                        });
+                    for start_pos in &self.state.start_pos[offset] {
+                        // check if length is sufficient
+                        if index + 1 - start_pos > self.finder.min_len {
+                            // build results
+                            self.state.found.push_back(Orf {
+                                start: start_pos - 2,
+                                end: index + 1,
+                                offset: offset as i8,
+                            });
+                        // if the first orf is too short, so are the others
+                        } else {
+                            break;
+                        }
```

Comment:
When a stop codon is found, loop through and queue all of the ORFs that can be constructed from the different start sites. Since these start sites are ordered from first to last (as they appear in the sequence), we can assume that if the ORF constructed from one start site is too small, then all of the other start sites will yield an even smaller, also invalid ORF. To save work in these cases, we break early from the loop.